lib/nexmo-jwt/jwt_builder.rb

```diff
@@ -0,0 +1,106 @@
+require 'openssl'
+require_relative 'jwt'
+
+module Nexmo
+  class JWTBuilder
+    # Generate an encoded JSON Web Token.
+    #
+    # By default the Nexmo JWT generator creates a short lived (15 minutes) JWT per request.
+    #
+    # To generate a long lived JWT for multiple requests, specify a longer value in the `exp`
+    # parameter in the `builder` block. 
+    #
+    # Example with no custom configuration:
+    #
+    # @example
+    #   token = Nexmo::JWTBuilder.build(application_id = YOUR_APPLICATION_ID, private_key = YOUR_PRIVATE_KEY)
+    #
+    # Example using the `builder` block to provide custom configuration options:
+    #
+    # @example
+    #   token = Nexmo::JWTBuilder.build(application_id = YOUR_APPLICATION_ID, private_key = YOUR_PRIVATE_KEY) do |builder|
+    #     builder.set_exp 500
+    #     builder.set_acl_paths({
+    #       "acl": {
+    #         "paths": {
+    #           "/messages": {
+    #             "methods": ["POST", "GET"],
+    #             "filters": {
+    #               "from": "447977271009"  
+    #             }
+    #           }
+    #         }
+    #       }
+    #     })
+    #     builder.set_sub 'My_Custom_Subject'
+    #   end
+    #
+    # @param [String] application_id
+    # @param [Integer] iat
+    # @param [SecureRandom::UUIDv4] jti
+    # @param [Integer] nbf
+    # @param [Integer] exp
+    # @param [Hash] paths
+    # @param [String] sub
+    # @param [String, OpenSSL::PKey::RSA] private_key
+    #
+    # @return [String]
+    #
+    def self.build(application_id, private_key)
+      builder = new
+      builder.set_application_id(application_id)
+      builder.set_private_key(private_key)
+      builder.set_alg
+      yield(builder) if block_given?
+      builder.jwt
+    end
+
+    def initialize
+      @jwt = Nexmo::Jwt.new
+    end
+
+    def set_application_id(application_id)
+      raise ArgumentError, "Missing required 'application_id' parameter" unless application_id
+
+      @jwt.application_id = application_id
+    end
+
+    def set_private_key(private_key)
+      raise ArgumentError, "Missing required 'private_key' parameter" unless private_key
+
+      if File.exist?(private_key)
+        @jwt.private_key = OpenSSL::PKey::RSA.new(File.read(private_key))
+      else
+        @jwt.private_key = OpenSSL::PKey::RSA.new(private_key)
+      end
+    end
+
+    def set_jti(jti)
+      @jwt.jti = jti
+    end
+
+    def set_nbf(nbf)
```

Comment:
This (and most of the other setter methods) should be using the machine names to make it easier to understand what each field is. So `set_nbf` should be something like `set_not_before`. Each field may have some data validation as well, like `set_ttl` should be within a certain range.